pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -592,6 +592,22 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext, dest destinationBitset) {
 	ReportDataSecAtRestInfo.Status, ReportDataSecAtRestInfo.Info =
 		vault.GetOperationalInfo(log)
 	ReportDeviceInfo.DataSecAtRestInfo = ReportDataSecAtRestInfo
+	if len(ReportDataSecAtRestInfo.VaultList) > 0 {
+		// We assume we care about the first one since normally only one
+		first := ReportDataSecAtRestInfo.VaultList[0]
+		var firstErr string
+		if first.VaultErr != nil {
+			firstErr = first.VaultErr.Description
+		}
+		if ctx.vaultStatus != first.Status ||
```

Comment:
As it is the only place where we initialize the state-related fields in the zedagent context, and the values depend on what we receive in the VaultStatus, shouldn't we wait for the first VaultStatus is received by the ZedAgent?
Something similar to what we have here:
https://github.com/lf-edge/eve/blob/e6c396103577ef486001ec3d99072c2b873ea37d/pkg/pillar/cmd/zedagent/zedagent.go#L458

Reply:
I'd have to check when vaultmgr runs and does the publish so that there isn't a signficant delay or deadlock.

Reply:
If using "uninitialized" values is acceptable, that's fine.